### PR TITLE
:book: Fix references to spec.kubernetesVersion for the Machine object to spec.Version

### LINKED ISSUE #3257: References to Machine spec.kubernetesVersion seems misleading
**What steps did you take and what happened:**
In the following files we reference `spec.kubernetesVersion` for the Machine object. But I don't believe we have that property. 

https://github.com/kubernetes-sigs/cluster-api/blob/fd73b181efeb3f2027c26841e1c08f5cab0322e9/bootstrap/kubeadm/types/v1beta1/types.go#L67

https://github.com/kubernetes-sigs/cluster-api/blob/fd73b181efeb3f2027c26841e1c08f5cab0322e9/bootstrap/kubeadm/config/crd/bases/bootstrap.cluster.x-k8s.io_kubeadmconfigs.yaml#L285

https://github.com/kubernetes-sigs/cluster-api/blob/fd73b181efeb3f2027c26841e1c08f5cab0322e9/bootstrap/kubeadm/config/crd/bases/bootstrap.cluster.x-k8s.io_kubeadmconfigs.yaml#L1091

https://github.com/kubernetes-sigs/cluster-api/blob/fd73b181efeb3f2027c26841e1c08f5cab0322e9/bootstrap/kubeadm/config/crd/bases/bootstrap.cluster.x-k8s.io_kubeadmconfigtemplates.yaml#L306

https://github.com/kubernetes-sigs/cluster-api/blob/fd73b181efeb3f2027c26841e1c08f5cab0322e9/bootstrap/kubeadm/config/crd/bases/bootstrap.cluster.x-k8s.io_kubeadmconfigtemplates.yaml#L1144

**What did you expect to happen:**
I expected it to be `spec.Version` as described here:
https://github.com/kubernetes-sigs/cluster-api/blob/fd73b181efeb3f2027c26841e1c08f5cab0322e9/api/v1alpha3/machine_types.go#L58-L61

/kind bug
/area api

### PR DESCRIPTION
**Pull request by saimanoj01**
**What this PR does / why we need it**:
Change reference from "spec.kubernetesVersion" to "spec.Version" for the Machine object.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3257

### CODE REVIEW

File: bootstrap/kubeadm/types/v1beta1/types.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ type ClusterConfiguration struct {
 	Networking Networking `json:"networking,omitempty"`
 
 	// KubernetesVersion is the target version of the control plane.
-	// NB: This value defaults to the Machine object spec.kubernetesVersion
+	// NB: This value defaults to the Machine object spec.Version
```
Comment (ncdc) [inline bootstrap/kubeadm/types/v1beta1/types.go:67]:
```suggestion
	// NB: This value defaults to the Machine object spec.version
```

**Comment by ncdc** [comment]
/ok-to-test
/approve

@saimanoj01 please let us know if you need any help resolving the CLA bot. Thanks!

**Comment by saimanoj01** [comment]
@ncdc Updated the email id on the commit with which I signed the CLA.

**Comment by ncdc** [comment]
Need to regen the yaml

**Comment by fabriziopandini** [comment]
@saimanoj01 please run `make generate` (or simply `make generate-manifests`) locally and push the regeneated yaml files as part of this change

**Pull request merged**